### PR TITLE
Feat gw/jt808

### PR DESCRIPTION
**Pull request by JimMoen**
Fixes [EMQX-3594](https://emqx.atlassian.net/browse/EMQX-3594)

### <samp>🤖 Generated by Copilot at f997419</samp>

This pull request adds support for the JT/T 808 gateway in EMQX, improves code consistency and readability for the GBT32960 gateway, and makes some code style and documentation improvements for the CoAP and ExProto gateways. It also modifies some test and configuration files to accommodate the new and existing gateway modules.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

File: apps/emqx_gateway_jt808/README.md [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# emqx_coap
```
Comment (HJianBo) [inline apps/emqx_gateway_jt808/README.md:1]:
```suggestion
# emqx_gateway_jt808
```

File: apps/emqx/test/emqx_proper_types.erl [hunk 2]
```diff
@@ -509,15 +509,16 @@ peercert() ->
...
-        emqx_lwm2m_protocol,
-        emqx_gbt32960_conn,
-        emqx_jt808_connection,
-        emqx_tcp_connection
+        %% emqx_coap_mqtt_adapter,
+        %% emqx_sn_gateway,
+        %% emqx_lwm2m_protocol,
+        %% emqx_gbt32960_conn,
+        %% emqx_tcp_connection
+        emqx_jt808_connection
```
Comment (HJianBo) [inline apps/emqx/test/emqx_proper_types.erl:521]:
```suggestion
```

File: apps/emqx_gateway_jt808/src/emqx_jt808_schema.erl [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+-include_lib("hocon/include/hoconsc.hrl").
+-include_lib("typerefl/include/types.hrl").
+
+-export([fields/1, desc/1]).
+
+-define(NOT_EMPTY(MSG), emqx_resource_validator:not_empty(MSG)).
+
+fields(jt808) ->
+    [
+        {frame, sc(ref(jt808_frame))},
```
Comment (HJianBo) [inline apps/emqx_gateway_jt808/src/emqx_jt808_schema.erl:17]: Miss the `mountpoint` option

Comment (JimMoen) [reply apps/emqx_gateway_jt808/src/emqx_jt808_schema.erl:17]: should the mountpoint be support to all gateway app?

Comment (JimMoen) [reply apps/emqx_gateway_jt808/src/emqx_jt808_schema.erl:17]: Mountpoint tested. 🚀

**Review by HJianBo — APPROVED** [review]
LGTM

**Pull request merged**